packages/suite/src/components/suite/PrerequisitesGuide/PrerequisitesGuide.tsx

```diff
@@ -44,71 +26,87 @@ export const PrerequisitesGuide = ({ showDeviceImage = true }: PrerequisitesGuid
-                    return <DeviceRecoveryMode />;
-                case 'device-initialize':
-                    return <DeviceInitialize />;
-                case 'device-bootloader':
-                    return <DeviceBootloader />;
-                case 'firmware-missing':
-                    return <DeviceNoFirmware />;
-                case 'firmware-required':
-                    return <DeviceUpdateRequired />;
-                case 'multi-share-backup-in-progress':
-                    return <MultiShareBackupInProgress />;
-                default:
-                    return <></>;
-            }
-        },
-        [prerequisite],
+    const deviceStatus = (device && getStatus(device)) ?? null;
+    const selectedDevice = useSelector(selectSelectedDevice);
+    const selectedDeviceModelInternal = selectedDevice
+        ? getDeviceInternalModel(selectedDevice)
+        : DEFAULT_FLAGSHIP_MODEL;
+
+    const showWarning =
+        !!(device && deviceStatus && deviceNeedsAttention(deviceStatus)) ||
+        prerequisite === 'no-transport';
+    const showWarningIcon = shouldDisplayInitialWarningIcon(deviceStatus);
+
+    const texts = getMessageId({
+        connected: !!device,
+        showWarning: showWarningIcon ?? showWarning,
+        deviceStatus,
+        prerequisite,
+    });
+
+    const DeviceImage = () => {
+        if (!showDeviceImage) return null;
+
+        return (
+            <Box margin={{ top: 40, bottom: 60 }}>
+                <DeviceWithScene
+                    deviceModel={selectedDeviceModelInternal}
+                    unitColor={selectedDevice?.features?.unit_color}
+                    height={300}
+                />
+            </Box>
+        );
+    };
+
+    const Heading = () => (
+        <Text typographyStyle="titleMedium" textWrap="balance" align="center">
+            <Translation id={texts.heading} />
+        </Text>
     );
 
-    const deviceStatus = (device && getStatus(device)) ?? null;
+    const Description = () => {
+        if (!texts.description) return null;
+
+        return (
+            <Text variant="tertiary" align="center" margin={{ top: 12 }}>
+                {texts.description}
+            </Text>
+        );
+    };
+
+    const TopAnimation = ({ children }: { children: React.ReactNode }) => (
+        <motion.div
+            initial={{ opacity: 0, y: -50 }}
+            animate={{ opacity: 1, y: -0 }}
+            transition={{ delay: 0.2, duration: 0.4, ease: motionEasing.enter }}
+            data-testid="@connect-device-prompt"
+        >
+            {children}
+        </motion.div>
+    );
+
+    const BottomAnimation = ({ children }: { children: React.ReactNode }) => (
+        <motion.div
+            initial={{ opacity: 0 }}
+            animate={{ opacity: 1 }}
+            transition={{ delay: 0.6, duration: 0.5, ease: motionEasing.enter }}
+        >
+            {children}
+        </motion.div>
+    );
 
     return (
```

Comment:
Everything is in one place. It's much more readable and it's super easy to update gaps between sections